src/legacy/ui/public/filter_bar/query_filter.js

```diff
@@ -18,19 +18,29 @@
  */
 
 import _ from 'lodash';
+import { Subject } from 'rxjs';
+
 import { onlyDisabled } from './lib/only_disabled';
 import { onlyStateChanged } from './lib/only_state_changed';
 import { uniqFilters } from './lib/uniq_filters';
 import { compareFilters } from './lib/compare_filters';
-import { EventsProvider } from '../events';
 import { mapAndFlattenFilters } from './lib/map_and_flatten_filters';
 import { extractTimeFilter } from './lib/extract_time_filter';
 import { changeTimeFilter } from './lib/change_time_filter';
 
-export function FilterBarQueryFilterProvider(Private, Promise, indexPatterns, $rootScope, getAppState, globalState, config) {
-  const EventEmitter = Private(EventsProvider);
+export function FilterBarQueryFilterProvider(Promise, indexPatterns, $rootScope, getAppState, globalState, config) {
+  const queryFilter = {};
+
+  const update$ = new Subject();
+  const fetch$ = new Subject();
+
+  queryFilter.getUpdates$ = function () {
+    return update$.asObservable();
```

Comment:
Is it necessary to convert subject to observable using `.asObservable()`?

Seems like this should work, too.

```js
queryFilter.getUpdates$ = function () {
  return update$;
};
```

If that is the case, maybe it could be simplified to

```js
queryFilter.update$ = new Subject();
```

and then from Angular subscribe using

```js
queryFilter.update$.subscribe(this.queryBarUpdateHandler)
```

Reply:
Used according to https://stackoverflow.com/a/36989035/372086
In order not to expose Subjects.

Also, update$ and fetch$ are not public on the returned object.